### PR TITLE
Add defaultPaymentInstrumentId and method to Customer and paymentInstrument to Payment  (GWY-4262, GWY-4319)

### PR DESCRIPTION
**Pull request by maxyengine**
Implements discriminator for payment instrument

Resolved: GWY-4262

### CODE REVIEW

File: spec/definitions/Customer.yaml [hunk 1]
```diff
@@ -20,10 +23,14 @@ properties:
...
     format: ipv4
-  defaultCardId:
-    description: The customer default card ID
-    allOf:
-      - $ref: "#/definitions/ResourceId"
+  defaultPaymentInstrument:
+    description: The customer's payment instrument
+    type: string
+    enum:
+      - PaymentCardPaymentInstrument
```
Comment (XaosSintez) [inline spec/definitions/Customer.yaml]: This is not correct case for enum values. It should be like `payment-card`

File: spec/definitions/PaymentInstument/AchPaymentInstrument.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+description: ACH payment instrument object
+allOf:
+  -
+    $ref: "#/definitions/Customer"
+  -
+    type: object
+    required:
+      - method
+      - bankAccountId
+    properties:
```
Comment (XaosSintez) [inline spec/definitions/PaymentInstument/AchPaymentInstrument.yaml]: Are this properties in root? Or in something like `paymentInstrument` property?

Comment (maxyengine) [inline spec/definitions/PaymentInstument/AchPaymentInstrument.yaml]: yes, i'll fix it

File: spec/definitions/PaymentInstument/cash.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+description: Cash payment instrument object
+allOf:
+  -
+    $ref: "#/definitions/PaymentInstrument"
+  -
+    type: "object"
```
Comment (slavcodev) [inline spec/definitions/PaymentInstument/cash.yaml]: that is unnecessary

File: spec/definitions/PaymentInstument/payment-card.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+description: Payment card payment instrument object
+allOf:
+  -
+    $ref: "#/definitions/PaymentInstrument"
+  -
+    type: "object"
+    required:
+      - paymentCardId
+    properties:
+      paymentCardId:
```
Comment (slavcodev) [inline spec/definitions/PaymentInstument/payment-card.yaml]: wonder, shouldn't be also `gatewayAccountId` in this instrument?

Comment (maxyengine) [inline spec/definitions/PaymentInstument/payment-card.yaml]: yes, i'll add it

File: spec/definitions/PaymentInstument/ach.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+description: ACH payment instrument object
+allOf:
+  -
+    $ref: "#/definitions/PaymentInstrument"
+  -
```
Comment (slavcodev) [inline spec/definitions/PaymentInstument/ach.yaml]: missing `type` here and in all other instruments

File: spec/definitions/PaymentInstument/paypal.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+description: PayPal payment instrument object
+allOf:
+  -
+    $ref: "#/definitions/PaymentInstrument"
+  -
+    type: "object"
+    required:
+      - payPalAccountId
+    properties:
```
Comment (adamaltman) [inline spec/definitions/PaymentInstument/paypal.yaml]: shouldn't this one also have `gatewayAccountId` as optional?

Comment (maxyengine) [inline spec/definitions/PaymentInstument/paypal.yaml]: we don't use it currently, but i'll add for future as option

File: spec/definitions/PaymentInstument/cash.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+description: Cash payment instrument object
+allOf:
+  -
+    $ref: "#/definitions/PaymentInstrument"
```
Comment (slavcodev) [inline spec/definitions/PaymentInstument/cash.yaml]: missing `receivedBy`

**Comment by slavcodev** [comment]
👍

**Pull request merged**